src/main/java/io/cdap/plugin/table/streaming/TableStreamingSourceConfig.java

```diff
@@ -0,0 +1,180 @@
+    "created with this schema, which will allow the table to be explored through CDAP.")
+  @Nullable
+  private String schema;
+
+  @Name(ROW_FIELD)
+  @Description("Optional schema field whose value is derived from the Table row instead of from a Table column. " +
+    "The field name specified must be present in the schema, and must not be nullable.")
+  @Nullable
+  private String rowField;
+
+  @Name(REFRESH_INTERVAL)
+  @Description("How often the table contents should be refreshed. Must be specified by a number followed by " +
+    "a unit where 's', 'm', 'h', and 'd' are valid units corresponding to seconds, minutes, hours, and days. " +
+    "Defaults to '1h'.")
+  @Nullable
+  private String refreshInterval;
+
+  public TableStreamingSourceConfig(String name, @Nullable String schema, @Nullable String rowField,
+                                    @Nullable String refreshInterval) {
+    this.name = name;
+    this.schema = schema;
+    this.rowField = rowField;
+    this.refreshInterval = refreshInterval;
+  }
+
+  private TableStreamingSourceConfig(Builder builder) {
+    this.name = builder.name;
+    this.schema = builder.schema;
+    this.rowField = builder.rowField;
+    this.refreshInterval = builder.refreshInterval;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(TableStreamingSourceConfig copy) {
+    return builder()
+      .setName(copy.name)
+      .setSchema(copy.schema)
+      .setRowField(copy.rowField)
+      .setRefreshInterval(copy.refreshInterval);
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  @Nullable
+  public String getRowField() {
+    return rowField;
+  }
+
+  public Schema getSchema() {
+    try {
+      return Schema.parseJson(schema);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Unable to parse the schema. Reason: " + e.getMessage());
+    }
+  }
+
+  public long getRefreshInterval() {
+    return refreshInterval == null ? TimeParser.parseDuration("1h") : TimeParser.parseDuration(refreshInterval);
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    Schema schema = getSchema();
+    if (!Strings.isNullOrEmpty(rowField) && schema.getField(rowField) == null) {
+      failureCollector.addFailure(String.format("Row field '%s' must be present in the schema", getRowField()), null)
+        .withConfigProperty(ROW_FIELD);
+    }
+
+    if (!Strings.isNullOrEmpty(refreshInterval)) {
+      String durationStr = refreshInterval.trim().toLowerCase();
+      String value = durationStr.substring(0, durationStr.length() - 1);
+
+      try {
+        Long.parseLong(value);
+      } catch (NumberFormatException var5) {
+        failureCollector.addFailure(
+          String.format("Error parsing the refresh interval '%s'. Cannot parse '%s' as a long.", durationStr, value),
+          "Provide correct refresh interval")
```

Comment:
correctiveAction does not add any new info. Hence should be `null`.

Reply:
Done.